storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -325,17 +328,25 @@ ssize_t Redo_Log_Reader::read_logfile(bool is_last, bool *finished) {
     const lsn_t end_lsn =
         read_log_seg(log, log_buf + len, start_lsn, start_lsn + RECV_SCAN_SIZE);
 
+    if (end_lsn == 0) {
+      xb::error() << "read_log_seg() failed.";
+      m_error = true;
+      return (-1);
+    }
+
     auto size = scan_log_recs(log_buf + len, is_last, start_lsn, &scanned_lsn,
                               log_scanned_lsn, finished);
-    if (end_lsn == start_lsn) {
-      break;
-    }
 
-    if (size < 0 || end_lsn == 0) {
-      xb::error() << "read_logfile() failed.";
+    if (size < 0) {
```

Comment:
isn't end_lsn = 0 an error?

Reply:
yes it is error. Just moved it before calling `scan_log_recs`